sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedByIteratorBase.java

```diff
@@ -91,7 +91,7 @@ synchronized void requestPage() {
                 this.done = continuationState.isDone();
 
                 return page;
-            }).blockLast();
+            }).blockFirst();
```

Comment:
Doing this seems safe to me.

Even though we allow  single call of `PageRetriever::get` to produce subset of pages, in almost all cases such a single call translated to single HTTP call that return single page. I mean `blockFirst` totally fine in this case.

In the case where the `get` produces subset of pages, I believe it safe to take the first page and drop the rest, provided the nextLink in that first page can be used to identify the correct next page, i.e. the second page in the dropped set.
This is an assumption and behavior we expect from a paginated endpoint and our existing `byPage(ct)` is built on this assumption.

Additionally the advanced cases where `PageRetriever::get` produce subset of pages,  if the production is lazy then unsubscribing after first has less impact. But for the cases where the subset is computed entirely before emitting, there is a cost involved in re-computation when second page is requested followed by the drop. Do you think by setting the second parameter of `PageRetriever::get` to 1 give implementor a chance to avoid unnecessary computation?

Reply:
Just a thought; didn't evaluated deeply but just wanted to share it. Basically the idea is:

1. We call PageRetriever::get that returns `Flux<Page>` that lazy produce subset of pages.
2. We convert this `Flux<Page>` to `Iterable` by calling `toIterable(batchSize:1)`.
3. We track this `Iterable`.
3. Then when `requestPage()` is called:  
4.1 we check `Iterable::hasNext` if so get the page `Iterable::next` 
4.2 if the `Iterable` is drained we go back to 1.

Not sure its too complicated but I was thinking what we can do to not to drop the computation flow of subset. Feel free to ignore if you find it over engineering :)

Reply:
Personally I think it might not be an appropriate assumption. The underlying PageRetriever do returns a `Flux`, and this implementation will discard anything other than the first item. We can assume mostly we will get `Mono.toFlux()`, but nothing to guarantee that.

BTW, I think the parameter of `pageSize` is number of items in a page, not the number of pages to retrieve - `@param pageSize The number of items to retrieve per page, passing {@code null} will use the source's default page size`.

Reply:
Just a thought; didn't evaluated deeply but just wanted to share it. Basically the idea is:

1. We call PageRetriever::get that returns `Flux<Page>` that lazy produce subset of pages.
2. We convert this `Flux<Page>` to `Iterable` by calling `toIterable(batchSize:1)`.
3. We track this `Iterable`.
3. Then when `requestPage()` is called:  
4.1 we check `Iterable::hasNext` if so get the page `Iterable::next` 
4.2 if the `Iterable` is drained we go back to 1.

Not sure its too complicated but I was thinking what we can do to not to drop the computation flow of subset. Feel free to ignore if you find it over engineering :)

Reply:
Tried this in https://github.com/Azure/azure-sdk-for-java/pull/18970

Currently the problem is that `Iterator` in reactor does one more request than neccessary.